packages/next/server/web/adapter.ts

```diff
@@ -103,6 +103,20 @@ export async function adapter(params: {
   const event = new NextFetchEvent({ request, page: params.page })
   let response = await params.handler(request, event)
 
+  // check if response is a Response object
+  if (
+    response &&
+    !(
+      typeof response === 'object' &&
+      (!response.headers || typeof response.headers.get === 'function') &&
+      (!response.body || typeof response.body.getReader === 'function') &&
+      typeof response.status === 'number' &&
+      typeof response.statusText === 'string'
+    )
```

Comment:
```suggestion
    !(response instanceof Response)
```

Can't we just check the instance?

Reply:
I decided not rely on the instance type -- to return a type which just behave like Response.
We have atleast to check if the header is the right type --- and body should be a stream.

Reply:
So I could theoretically just return:
```ts
{
  headers: {
    get() {}
  },
  body: {
    getReader() {}
  },
  status:  1000,
  statusText: "Hi
}
```

I feel that for conformance, we need to be stricter than this. :thinking: